languages/python/oso/polar/polar.py

```diff
@@ -120,6 +131,7 @@ def load_str(self, string, filename=None):
 
     def clear_rules(self):
         self.ffi_polar.clear_rules()
+        self._polar_roles_enabled = False
```

Comment:
Hm, didn't think about this. Should we auto-reload the Polar roles policy if it was enabled with `clear_rules` was called? Or is it better to require the user to explicitly re-enable? I think from the user's perspective, they might not understand that "enabling roles" involves loading rules, so wouldn't realize that a side-effect of `clear_rules` is deactivating roles. But I don't really know when people use `clear_rules` anyway.

Reply:
That's a good question. Will think more about it.

Reply:
> Should we auto-reload the Polar roles policy if it was enabled with clear_rules was called? ... I think from the user's perspective, they might not understand that "enabling roles" involves loading rules, so wouldn't realize that a side-effect of clear_rules is deactivating roles.

Agree with this reasoning; implemented in 5ba7d14.

Only downside is that calling `clear_rules()` won't leave you with a completely empty KB, which might be unexpected, but as you said I'm not sure folks would realize `enable_roles()` loads rules unless they looked at the code or trace output.